File: sdks/ruby/lib/novabilling/add_ons/types/create_add_on_dto.rb
# frozen_string_literal: true

module Novabilling
  module AddOns
    module Types
      class CreateAddOnDto < Internal::Types::Model
        field :name, -> { String }, optional: false, nullable: false
        field :code, -> { String }, optional: false, nullable: false
        field :description, -> { String }, optional: true, nullable: false
        field :invoice_display_name, -> { String }, optional: true, nullable: false, api_name: "invoiceDisplayName"
        field :prices, -> { Internal::Types::Array[Novabilling::Types::AddOnPriceDto] }, optional: false, nullable: false
        field :created_at, -> { String }, optional: true, nullable: false, api_name: "createdAt"
      end
    end
  end
end

File: sdks/rust/src/api/types/create_add_on_dto.rs
pub use crate::prelude::*;

/// Request type for API operation
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct CreateAddOnDto {
    /// Display name
    pub name: String,
    /// Unique code for the add-on
    pub code: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    /// Custom name shown on invoices
    #[serde(rename = "invoiceDisplayName")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub invoice_display_name: Option<String>,
    /// Prices in different currencies
    pub prices: Vec<AddOnPriceDto>,
    /// Backdate createdAt (ISO 8601). For data imports.
    #[serde(rename = "createdAt")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_at: Option<String>,
}
